### ISSUE TITLE
suffix of .in-addr-arpa added cumulatively to all records when editing PTR zone

### ISSUE THREAD

**Issue by falz** [PowerDNS-Admin/PowerDNS-Admin#957]
A bug seems to exist for IPV4 PTR zones. Our zones are generally /24 in size. 

If one goes to the zone and edits one record, enters something valid such as "2". When you save and apply, it auto adds the suffix of <zone>.in-addr-arpa to all other records on the zone. The record you just edited is fine.

If you continue to edit other records, it will keep adding this for the records you did not edit.

This does not occur if you have Auto PTR enabled. We cannot enable Auto PTR due to a different bug (https://github.com/ngoduykhanh/PowerDNS-Admin/issues/640)

We're currently running from git master where last commit was 4c19f959285d125437555aebace903640f3f652c (per `git log`) which was January 2021, so not fully up to date.

![Screenshot from 2021-06-10 13-45-32](https://user-images.githubusercontent.com/707319/121580693-a60bec80-c9f2-11eb-9328-a97e3ef1986f.png)

**Comment by falz** [comment]
As of this morning i updated to current (git pull), performed proper db migration stuff, issue still persists.

**Comment by falz** [comment]
Another note. I inspected the database, and it adds weird half records when this issue occurs. All of the types where it's NULL, and the 'name' strangely adds one for each dot in the zone name. ie if zone is 1.2.3.in-addr.arpa, it adds

in-addr.arpa
3.in-addr.arpa
2.3.in-addr.arpa
1.2.3.in-addr.arpa

![Screenshot from 2021-06-16 06-03-40](https://user-images.githubusercontent.com/707319/122210880-bd702d00-ce6b-11eb-9c1a-5f4c4dc680a6.png)

**Comment by falz** [comment]
Spoke to developers of powerdns, the issue above with null values is NOT an issue and is expected, so that can be ignored.

However, I just did a from scratch fresh install of powerdns-admin from git, dated today, and this issue with PTR's still occurs. 

I'd love to hear if anyone does *not* have this issue and is running a current-ish version?

**Comment by steschuser** [comment]
I'll try to take a look at it this week.

Can you write down the steps to reproduce this from a fresh install?

**Comment by falz** [comment]
This issue was discovered when testing on our live zones (approx 1k) when using powerdns's `zone2sql` command. I tested by creating a PTR zone from scratch, using powerdns-admin's web ui, and did not have the issue.

The difference seems to be that imported zones from `zone2sql` use uppercase characters in the zone name:

* 1.168.192.IN-ADDR.ARPA

while if one creates from webui, even if you *type* uppercase, the zone that gets created is lowercase:

* 1.168.192.in-addr.arpa

After reviewing even further, it appears that the record types that one lets you enter on an UPPERCASE reverse zone.. aren't actually for a reverse zone (A, AAAA, MX..)

The bug appears to be that PowerDNS-Admin does not properly 'sense' a PTR zone if uppercase characters are used anywhere in the zone name (tried all UPPERCASE, and some letters UpperCase)

Powerdns itsself serves these zones just fine when it has uppercase characters, so this appears to be a bug with Powerdns-Admin.

The fix appears to require at least two things.

1) detect zone as type 'reverse' which limits the record types one can add. In `routes/domain.py`, i tossed in **, re.IGNORECASE** and it seemed to do this.

```
    if not re.search(r'ip6\.arpa|in-addr\.arpa$', domain_name, re.IGNORECASE):
        editable_records = forward_records_allow_to_edit
    else:
        editable_records = reverse_records_allow_to_edit
```

2) Once you are editing the zone, some type of detection is being missed that each of these records is for a reverse zone, and the 'Name' column shows as my first screenshot. I tried to add a few **re.IGNORECASE**'s in to `models/domain.py` in this function, but no avail.

```
    def get_reverse_domain_name(self, reverse_host_address):
       < a few re.searches related to arpa are here >
```
